src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/SqlThrottlingCondition.cs

```diff
@@ -0,0 +1,337 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Data.SqlClient;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data.SqlClient.Reliability.Data
+{
+    /// <summary>
+    /// Defines the possible throttling modes in SQL Database.
+    /// </summary>
+    public enum ThrottlingMode
+    {
+        /// <summary>
+        /// Corresponds to the "No Throttling" throttling mode, in which all SQL statements can be processed.
+        /// </summary>
+        NoThrottling = 0,
+
+        /// <summary>
+        /// Corresponds to the "Reject Update / Insert" throttling mode, in which SQL statements such as INSERT, UPDATE, CREATE TABLE, and CREATE INDEX are rejected.
+        /// </summary>
+        RejectUpdateInsert = 1,
```

Comment:
The name is incorrect. This option does not just affect updates and inserts but a lot more. Maybe use `Write`. How is this different from `RejectAllWrites`?

Reply:
This was the decoding of server-side throttling conditions, we'll look into it to check if still relevant.